ruby/red-gandiva/lib/gandiva/field-node.rb

```diff
@@ -0,0 +1,64 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+module Gandiva
+  class FieldNode
+    def +(node)
+      build_function_node("add", node)
```

Comment:
We should not convert to `Gandiva::Node` here.
If we convert to `Gandiva::Node` here, we can't combine more expression: `record.field1 + record.field2 + record.field3`

We should keep internal builder objects while evaluating expression builder block (the block passed to `Arrow::Schema#build_expression`). We should convert the internal builder objects after evaluation expression builder block:

```ruby
module Arrow
  class Schema
    def build_expression(&block)
      builder = Gandiva::ExpressionBuilder.new(self)
      builder.build(&block)
    end
  end
end

module Gandiva
  class ExpressionBuilder
    def initialize(schema)
      @schema = schema
    end

    def build
      # Not node = yield(...)
      builder = yield(Record.new(@schema), Context.new(@schema))
      builder.build
    end
  end
end
```